lib/fastly/client.rb

```diff
@@ -102,6 +102,11 @@ def purge(url, params = {})
       extras = params.delete(:headers) || {}
       uri    = URI.parse(url)
       http   = Net::HTTP.new(uri.host, uri.port)
+
+      if uri.is_a? URI::HTTPS
```

Comment:
Ah, I'll update that, then.

With further checking it turns out that `URI.parse` and the `Kernel::URI` methods don't behave quite the same.

```
$ ruby -ruri -e 'p URI URI "http://example"'
#<URI::HTTP http://example>
$ ruby -ruri -e 'p URI.parse URI "http://example"'
/usr/local/lib/ruby/2.3.0/uri/rfc3986_parser.rb:18:in `rescue in split': bad URI(is not URI?): http://example (URI::InvalidURIError)
	from /usr/local/lib/ruby/2.3.0/uri/rfc3986_parser.rb:15:in `split'
	from /usr/local/lib/ruby/2.3.0/uri/rfc3986_parser.rb:73:in `parse'
	from /usr/local/lib/ruby/2.3.0/uri/common.rb:227:in `parse'
	from -e:1:in `<main>'
```

So I'll make passing a `URI` object in allowable in a followup commit to this one by changing `URI.parse` to `URI`